pkg/providers/vsphere/vmlifecycle/update_status.go

```diff
@@ -435,14 +435,17 @@ func UpdateNetworkStatusConfig(vm *vmopv1.VirtualMachine, args BootstrapArgs) {
 
 	// Update the global DNS information.
 	{
-		hn, ns, sd := args.Hostname, args.DNSServers, args.SearchSuffixes
-		lhn, lns, lsd := len(hn) > 0, len(ns) > 0, len(sd) > 0
-		if lhn || lns || lsd {
+		hn, dn, ns, sd := args.HostName, args.DomainName, args.DNSServers, args.SearchSuffixes
+		lhn, ldn, lns, lsd := len(hn) > 0, len(dn) > 0, len(ns) > 0, len(sd) > 0
+		if lhn || ldn || lns || lsd {
 			nc.DNS = &vmopv1.VirtualMachineNetworkConfigDNSStatus{}
 		}
 		if lhn {
```

Comment:
Maybe while here move these under the `lhn || ldn || lns || lsd` block so it clearer they are related, and maybe a little harder to forget to update the first check